### ISSUE TITLE
nvda does not read the group boxes in audacity

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3473]
**Reported by david on 2013-08-27 09:25**
This problem occurs when wxWidgets is compiled with the wxUSE_ACCESSIBILITY flag set to 1. This is needed so that the custom controls in audacity can be made accessible.
I discussed this issue with Jamie on devel mailing list, and the problem is that if a group box is present, it is a previous sibling of a control, rather than the previous sibling of the controls parent. Jamie suggested a work around in the audacity appmodule to set the container property.
Attached is a patch in the form of a diff. This has been tested both on the current audacity, and with a version of audacity where in effect the wxUSE_ACCESSIBILITY flag has been set to 0. So if the problems in accessibility code in wxWidgets are fixed at some point, this work around will not prevent the group boxes from being read.
Just a very minor point. Jamie suggested using an overlay class to set the container property. I tried this and the results are similar but not identical. Using the overlay class resulted in a minor bug: If the first control in a group box has a static text label which isn't the same as the accessibility name of the control, then when the control becomes the focus, nvda reads the name of the grouping, followed by the static text, followed by the name of the control.

**Comment by bhavyashah** [comment]
@jcsteh Your https://github.com/nvaccess/nvda/issues/3473#issuecomment-155309126 is slightly unclear to me. Do you consider the issue raised as universal and the patch to be app-specific, or is the submission ready for code review but one hasn't had the chance to do that yet, or something else? I would appreciate a clarification.

**Comment by jcsteh** [comment]
It's theoretically broader than one app (wx apps built with wxUSE_ACCESSIBILITY containing group boxes), but no one else has reported it for other apps and there's an app specific patch, so I don't see any harm in considering this for inclusion as app specific. It can always be generalised later. However, one problem is that we don't have contact with the reporter, so we can't get any issues addressed or testing performed, and this is not something we can prioritise doing ourselves. For now, I'm going to close this, but if someone wants to take this up, we can reopen and accept a PR.

One additional response:

> Jamie suggested using an overlay class to set the container property. I tried this and the results are similar but not identical. Using the overlay class resulted in a minor bug: If the first control in a group box has a static text label which isn't the same as the accessibility name of the control, then when the control becomes the focus, nvda reads the name of the grouping, followed by the static text, followed by the name of the control.

This is intentional. The idea is that this static text is generally the descriptive text for the group. If this is incorrect for Audacity, the correct fix is to override the description for groupings, setting it to None.
